urllib3/contrib/pyopenssl.py

```diff
@@ -127,6 +129,17 @@ def extract_from_urllib3():
     util.ssl_.IS_PYOPENSSL = False
 
 
+def _validate_dependencies_met():
+    """
+    Verifies that PyOpenSSL's package-level dependencies have been met.
+    Throws `ImportError` if they are not met.
+    """
+    # Method added in `cryptography==1.1`; not available in older versions
+    from cryptography.x509.extensions import Extensions
+    if not hasattr(Extensions, "get_extension_for_class"):
+        raise ImportError("'cryptography' module missing required function (try upgrading the module?): 'cryptography.x509.extensions.Extensions.get_extension_for_class'")
```

Comment:
@aseering, I definitely agree with checking for functionality rather than hardcoding a version check. @Lukasa will likely have a better guess as to what feature(s) make 1.3.4 the min requirement.